src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -607,8 +650,81 @@ private boolean isAllowedApiRequest(HttpServletRequest request) {
 			url=pathInfo;
 			isAllowedRequest=isApiFoundIntheList(url);
 		}
+
+		if( url.matches(appExtPropertiesFile.getProperty(CANVAS_API_CROSSLIST)) && session.getAttribute(LAUNCH_TYPE).equals("lti")){
+			isAllowedRequest = isCrosslistAllowed(request, session, url);
+		}
+
 		return isAllowedRequest;
 	}
+
+	private boolean isCrosslistAllowed(HttpServletRequest request,
+			HttpSession session, String url) {
+		boolean isSectionMatch = false;
+
+		//build api call
+		String crosslistApiCall = canvasURL + url.substring(0, url.indexOf("/crosslist"));
+		M_log.debug("crosslist API call: " + crosslistApiCall);
+
+		//String is built, time to make the call
+		String sessionId = request.getSession().getId();
+		String loggingApiWithSessionInfo = String.format("Canvas API request with Session Id \"%s\" for URL \"%s\"", sessionId,crosslistApiCall);
+		M_log.info(loggingApiWithSessionInfo);
+		HttpUriRequest clientRequest = null;
+
+		clientRequest = new HttpGet(crosslistApiCall);
+
+		HttpClient client = new DefaultHttpClient();
+		final ArrayList<NameValuePair> nameValues = new ArrayList<NameValuePair>();
+		nameValues.add(new BasicNameValuePair("Authorization", "Bearer"+ " " +canvasToken));
+		nameValues.add(new BasicNameValuePair("content-type", "application/json"));
+		for (final NameValuePair h : nameValues)
+		{
+			clientRequest.addHeader(h.getName(), h.getValue());
+		}
+		BufferedReader rd = null;
+		long startTime = System.currentTimeMillis();
+		try {
+			rd = new BufferedReader(new InputStreamReader(client.execute(clientRequest).getEntity().getContent()));
+		} catch (IOException e) {
+			M_log.error("Canvas API call did not complete successfully", e);
+		}
+		long stopTime = System.currentTimeMillis();
+		long elapsedTime = stopTime - startTime;
+		M_log.info(String.format("CANVAS Api response took %sms",elapsedTime));
+		String line = "";
+		StringBuilder sb = new StringBuilder();
+		try {
+			while ((line = rd.readLine()) != null) {
+				sb.append(line);
+			}
+		} catch (IOException e) {
```

Comment:
The code from 677-704 is Duplicated again. The same Code is repeated in the `getCanvasResponse()` in the class. Try to Refactor the common code used in the  `getCanvasResponse()` and `isCrossListAllowed()`. This will limit the Number of line in the code for readability and maintenance